api/tests/views_tests.py

```diff
@@ -34,34 +60,136 @@ def test_runtime_data(client):
+        "error_context": {"unavailable_address": "myNewAlias"},
+    }
+
+
+def test_post_domainaddress_free_user_error(free_api_client):
+    """A free user is not allowed to create a domain address."""
+    response = free_api_client.post(
+        reverse("domainaddress-list"), data={"address": "my-new-alias"}, format="json"
+    )
+
+    assert response.status_code == 403
+    ret_data = response.json()
+    assert ret_data == {
+        "detail": "You must be a premium subscriber to create subdomain aliases.",
+        "error_code": "free_tier_no_subdomain_masks",
+    }
+
+
+def test_post_relayaddress_success(settings, free_api_client) -> None:
+    """A free user is able to create a random address."""
+    response = free_api_client.post(
+        reverse("relayaddress-list"), data={}, format="json"
+    )
+
+    assert response.status_code == 201
+    ret_data = response.json()
+    assert ret_data["enabled"]
+
+
+def test_post_relayaddress_free_mask_email_limit_error(
+    settings, free_api_client
+) -> None:
+    """A free user is unable to exceed the mask limit."""
+    free_user = get_user(free_api_client)
     for _ in range(settings.MAX_NUM_FREE_ALIASES):
-        assert check_user_can_make_another_address(free_profile) is None
         baker.make(RelayAddress, user=free_user)
-        del free_profile.relay_addresses  # Invalidate cached_property
-    with pytest.raises(CannotMakeAddressException):
-        check_user_can_make_another_address(free_profile)
 
-    url = reverse("relayaddress-list")
-    client = APIClient()
-    client.force_authenticate(user=free_user)
-    origin = "https://login.example.com"
-    data = {
-        "enabled": True,
-        "description": origin,
-        "generated_for": origin,
-        "used_on": origin,
-    }
-    response = client.post(reverse("relayaddress-list"), data, format="json")
+    response = free_api_client.post(reverse("relayaddress-list"), {}, format="json")
+
     assert response.status_code == 400
+    ret_data = response.json()
+    assert ret_data == {
+        "detail": (
+            "You must be a premium subscriber to make more than"
+            f" {settings.MAX_NUM_FREE_ALIASES} aliases."
+        ),
+        "error_code": "free_tier_limit",
+        "error_context": {"free_tier_limit": 5},
+    }
 
-    ret_data = response.json()  # type: ignore[attr-defined]
-    expected_msg = (
-        "You must be a premium subscriber to make more than"
-        f" {settings.MAX_NUM_FREE_ALIASES} aliases."
-    )
-    assert ret_data == {"non_field_errors": [expected_msg]}
+
+def test_post_relayaddress_flagged_error(free_api_client) -> None:
+    """A flagged user is unable to create a random mask."""
+    free_profile = get_user(free_api_client).profile_set.get()
+    free_profile.last_account_flagged = timezone.now()
+    free_profile.save()
+
+    response = free_api_client.post(reverse("relayaddress-list"), {}, format="json")
+
+    assert response.status_code == 400
```

Comment:
suggestion (non-blocking): this one seems like it should be 403. i.e., if a different user made the same exact request, it could succeed.